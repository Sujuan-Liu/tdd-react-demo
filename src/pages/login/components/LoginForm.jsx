import { PureComponent } from "react";

class LoginForm extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      username: '',
      password: '',
      errorMsg: '',
    }
    this.handleInputChange = this.handleInputChange.bind(this);
  }

  handleInputChange = event => {
    /**
     * change state
     */
    this.setState({
      ...this.state,
      [event.target.name]: event.target.value
    });
  }

  handleSubmit = event => {
    event.preventDefault();
    const { username, password } = this.state;
    let errorMsg = '';
    if (!username || !password) {
      errorMsg = 'username or password is required';
      this.setState({
        ...this.state,
        errorMsg,
      });
      return;
    }
    // call father
    const { submitForm } = this.props;
    submitForm({ username, password });
  }

  render() {
    const { username, password, errorMsg } = this.state;
    return (
      <div>        
        <form>
          <p className="error-info">{errorMsg}</p>
          <input type="text" name="username" value={username || ''} onChange={this.handleInputChange} />
          <input type="password" name="password" value={password || ''} onChange={this.handleInputChange} />
          <button className="login" onClick={this.handleSubmit}>Login</button>
        </form>
      </div>
    );
  }
}

export default LoginForm;
