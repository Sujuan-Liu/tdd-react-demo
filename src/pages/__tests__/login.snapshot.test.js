import LoginForm from '../login/components/LoginForm';
import renderer from 'react-test-renderer';


describe('Page: login', () => {
  it('Render correctly', () => {
    const loginFormTree = renderer.create(<LoginForm />).toJSON();
    expect(loginFormTree).toMatchSnapshot();
  });
});