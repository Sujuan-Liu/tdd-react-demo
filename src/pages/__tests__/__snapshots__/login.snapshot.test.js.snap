// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: login Render correctly 1`] = `
<div>
  <form>
    <p
      className="error-info"
    >
      
    </p>
    <input
      name="username"
      onChange={[Function]}
      type="text"
      value=""
    />
    <input
      name="password"
      onChange={[Function]}
      type="password"
      value=""
    />
    <button
      className="login"
      onClick={[Function]}
    >
      Login
    </button>
  </form>
</div>
`;
